data/LocalStore.js

```diff
@@ -33,12 +35,25 @@ export class LocalStore extends BaseStore {
      * @param {Object} c - LocalStore configuration.
      * @param {function} [c.processRawData] - Function to run on data
      *      presented to loadData() before creating records.
+     * @param {[function|string]} [c.idSpec] - specification of how to identify an immutable unique
+     *      id for each record.  May be either a property (default is 'id') or a function to create the unique id
+     *      from the record.  If you cannot identify a unique id, you may set this argument to `XH.genId`
+     *      to have a unique id  generated on the fly for each record.  Note that in this case, grids and other
+     *      components bound to this store will not be able to maintain state for records across data reloading.
      * @param {function} [c.filter] - Filter function to be run.
      * @param {...*} [c.baseStoreArgs] - Additional properties to pass to BaseStore.
      */
-    constructor({processRawData = null, filter = null, ...baseStoreArgs}) {
+    constructor(
+        {
+            processRawData = null,
+            filter = null,
+            idSpec = 'id',
+            ...baseStoreArgs
+
```

Comment:
Blank line here

Reply:
Actually the Ctor of Recordset should be checking the unique -- but on closer examination, its busted.  Fixing now.